packages/esm-patient-search-app/src/patient-search-page/advanced-patient-search.component.tsx

```diff
@@ -0,0 +1,145 @@
+import React, { useEffect, useMemo, useState } from 'react';
+import { useGetPatientAttributePhoneUuid, usePatientSearchInfinite } from '../patient-search.resource';
+import { AdvancedPatientSearchState } from '../types';
+import styles from './advanced-patient-search.scss';
+import { initialState } from './advanced-search-reducer';
+import PatientSearchComponent from './patient-search-lg.component';
+import RefineSearch from './refine-search.component';
+
+interface AdvancedPatientSearchProps {
+  query: string;
+  inTabletOrOverlay?: boolean;
+  stickyPagination?: boolean;
+  selectPatientAction?: (patientUuid: string) => void;
+  hidePanel?: () => void;
+}
+
+const AdvancedPatientSearchComponent: React.FC<AdvancedPatientSearchProps> = ({
+  query,
+  stickyPagination,
+  selectPatientAction,
+  inTabletOrOverlay,
+  hidePanel,
+}) => {
+  const [filters, setFilters] = useState<AdvancedPatientSearchState>(initialState);
+  const filtersApplied = useMemo(() => {
+    let count = 0;
+    Object.entries(filters).forEach(([key, value]) => {
+      if (value != initialState[key]) {
+        count++;
+      }
+    });
+    return count;
+  }, [filters]);
```

Comment:
This seems like a good job for Array.reduce

Reply:
filters is actually an object, how can I use Array.reduce?

Reply:
I believe it would be `return Object.entries(filters).reduce((prev, ([key, value])) => prev + (value != initialState[key] ? 1 : 0))`